eudemo/eudemo/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+from bluemira.builders.tools import circular_pattern_component
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.plane import BluemiraFace, BluemiraPlane
+from bluemira.geometry.tools import (
+    boolean_fuse,
+    extrude_shape,
+    make_polygon,
+    slice_shape,
+)
+from bluemira.geometry.wire import BluemiraWire
+
+
+class EquatorialPort(ComponentManager):
+    """
+    Wrapper around a Equatorial Port component tree
+    """
+
+    def xz_boundary(self) -> BluemiraWire:
+        """Returns a wire defining the x-z boundary of the Equatorial Port"""
+        return (
+            self.component.get_component("xz")
+            .get_component(EquatorialPortBuilder.KOZ)
+            .shape.boundary[0]
+        )
+
+
+@dataclass
+class EquatorialPortDesignerParams(ParameterFrame):
+    """
+    Equatorial Port Designer parameters
+    """
+
+    ep_height: Parameter[float]
+
+
+@dataclass
+class EquatorialPortBuilderParams(ParameterFrame):
+    """
+    Equatorial Port builder parameters
+    """
+
+    n_ep: Parameter[int]
+    ep_r_corner: Parameter[float]
+
+
+class EquatorialPortDesigner(Designer):
+    """
+    Equatorial Port Designer
+    """
+
+    param_cls: Type[EquatorialPortDesignerParams] = EquatorialPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortDesignerParams],
+        build_config: Union[Dict, None],
+        x_ib: float,
+        x_ob: float,
+    ):
+        super().__init__(params, build_config)
+        self.x_ib = x_ib
+        self.x_ob = x_ob
+
+    def run(self) -> BluemiraWire:
+        """
+        Design the xz keep-out zone profile of the equatorial port
+        """
+        x_ib = self.x_ib
+        x_ob = self.x_ob
+        z_h = self.params.ep_height.value
+
+        x = (x_ib, x_ob, x_ob, x_ib)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class EquatorialPortBuilder(Builder):
+    """
+    Equatorial Port Builder
+    """
+
+    NAME_XY = "Equatorial Ports xy"
+    NAME_XYZ = "Equatorial Ports xyz"
+    KOZ = "koz"
+    param_cls: Type[EquatorialPortBuilderParams] = EquatorialPortBuilderParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortBuilderParams],
+        build_config: Union[Dict, None],
+        xz_profile: BluemiraFace,
+        y_width: float,
+        x_offsets: list,
+        castellation_offsets: list,
+    ):
+        super().__init__(params, build_config)
+        self.xz_profile = xz_profile
+        self.y_width = y_width
+        self.x_off = x_offsets
+        self.cst = castellation_offsets
+
+    def build(self) -> Component:
+        """
+        Build the equatorial port component.
+        """
+        # TODO: Implement corner radii
```

Comment:
#1992 should help with this, but dont worry about it now

Reply:
Cool, I'll look at this later